src/main/java/core/basesyntax/service/operations/FruitTransactionReportMakerImpl.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class FruitTransactionReportMakerImpl implements FruitTransactionReportMaker {
+    @Override
+    public List<FruitTransaction> makeReport(FruitDao dao) {
+        List<FruitTransaction> balanceList = dao.getFirst("b");
+        for (FruitTransaction currentFruitTransaction : balanceList) {
```

Comment:
it's not a strategy pattern. You should create additional service called OperationStrategy,  OperationHendler interface with one method `handle()`, and four realizations of this interface (according to the operation). OperationStrategy should contains map with FruitTransaction.Operation key and with OperationHandler value.